### ISSUE TITLE
import/no-cycle constantly throws false positives when a file has multiple exports

### ISSUE THREAD

**Issue by TidyIQ** [benmosher/eslint-plugin-import#1387]
For example, let's say I have these two files:

PageOne.js
```
import { bar } from "./PageTwo";

export const foo = "foo";

const PageOne = () => {
  return (
    <p>{bar}</p>
    <p>{foo}</p>
  )
};
export default PageOne;
```
PageTwo.js
```
import { foo} from "./PageOne ";

export const bar = "bar";

const PageTwo= () => {
  return (
    <p>{foo}</p>
    <p>{bar}</p>
  )
};
export default PageTwo;
```
There is no dependency cycle issue with this. Even though each module is importing from each other, they are not dependent on each other.

It seems this rule only checks the list of imports rather than checking what is actually imported and whether it has a cycle dependency. If two files import from each other but neither of the imported modules depend on each other then an error shouldn't occur for this rule. In the above example, the imported `bar` in PageOne.js doesn't depend on the value of `foo`, and the imported `foo` in PageTwo.js doesn't depend on the value of `bar`.

I just installed this plugin and I'm getting this error all throughout my code. I would prefer to keep the rule enabled for the times where there is a legitimate issue, but it seems I have no option but to turn this off.

**Comment by ljharb** [comment]
Yes, there is a cycle - cycles are based on the module, not the import.

**Comment by ljharb** [comment]
To avoid cycles, no module C can import any module/file that depends (directly or transitively) on X.

**Comment by benmosher** [comment]
Working as intended 👍🏼 the rule intends to let you know module-level cycles. It is not smart enough to check whether module-level code uses the imports from the cycle. That would be neat but this is rule is more intended to just let you know that your module dependency graph is or is not a directed acyclic graph.

Every rule is not for every project.

**Comment by FranciscoKloganB** [comment]
I am not quite getting the explanations or implementation of this rule. I am currently getting an error with the following files

`constants.ts`
`handleClear.ts`
`reducer.ts`

**Constants does not depend on anything**.
The **reducer consumes** N constants from **`constants`**
The **reducer consumes** a function from **`handleClear`**
The **handleClear** file **pulls a single constant** from `constants

```
// Rough graph
constants <------------- handleClear
   ^                                              ^
    |                                               |
    |                                               |
   reducer ---------------------
```

This is not a Cycle, because once you reach `constants` module you never get out it. It's an absorbent node. If somehow this rule identifies this as a cycle it's breaking basic rules of graphs. Because without this structure, the only option is to write hard-coded strings. 

Can someone elaborate?

**Comment by ljharb** [comment]
Can you share a codesandbox or repo with these files?
